boltdesignsystem/bolt
# Fix OL number bugs in Firefox

## Jira

http://vjira2:8080/browse/BDS-2028

## Summary

Fixes alignment and incrementing bugs on OL in Firefox.

## Details

See alignment and incrementing bugs:

![image](https://user-images.githubusercontent.com/36336/75278158-5c849780-57d7-11ea-9b5d-86a71d2c597f.png)

This appears to be caused by a recent Firefox update. These issues don't exist in older versions.

Setting the `bolt-li` to `display: block` fixes the number position issue. Custom elements are `inline` by default, and somehow that threw off the relative position in Firefox.

Number incrementing was fixed by removing an extra `number-increment` rule which applied only to list items in the shadow DOM. This was causing the counter to be reset for each encapsulated item (so always `1`). Again, not sure why this just started to be an issue in Firefox, but retested and now works cross-browser.

## How to test
> Note: bugs only occur in recent versions of Firefox. Test there and also check cross-browser for  regressions.
- View [OL demo on master](https://boltdesignsystem.com/pattern-lab/patterns/02-components-ol-15-ol-nested-items/02-components-ol-15-ol-nested-items.html) to see bugs.
- View OL demo on feature branch (waiting to deploy) and verify bugs are fixed.
